exporter/redis.go

```diff
@@ -711,27 +713,34 @@ func scanForKeys(c redis.Conn, pattern string) ([]string, error) {
 			return keys, fmt.Errorf("invalid response from SCAN for pattern: %s", pattern)
 		}
 
-		k, _ := redis.Strings(arr[1], nil)
+		k, err := redis.Strings(arr[1], nil)
+		if err != nil {
+			log.Debugln(err)
+		}
 		keys = append(keys, k...)
 
-		if iter, _ = redis.Int(arr[0], nil); iter == 0 {
+		if iter, err = redis.Int(arr[0], nil); iter == 0 {
 			break
 		}
+		if err != nil {
+			log.Debugln(err)
+		}
 	}
 
 	return keys, nil
 }
 
 // getKeysFromPatterns does a SCAN for a key if the key contains pattern characters
 func getKeysFromPatterns(c redis.Conn, keys []dbKeyPair) (expandedKeys []dbKeyPair, err error) {
+	var keyNames []string
```

Comment:
Same, why move this up?

Reply:
same here

Reply:
I'd remove the named return variables, make 736 an instantiation (`:=`) and then the shadowing issue should be resolved.